src/couch_mrview/src/couch_mrview_debug.erl

```diff
@@ -48,3 +78,36 @@ view_signature(DbName, DDocName) ->
     {ok, DDoc} = couch_db:open_doc_int(Db, <<"_design/", DDocName/binary>>, []),
     {ok, IdxState} = couch_mrview_util:ddoc_to_mrst(DDocName, DDoc),
     couch_util:to_hex(IdxState#mrst.sig).
+
+view_state(Pid) ->
+    {ok, IdxState} = couch_index:get_state(Pid, 0),
+    case IdxState of
+        #mrst{} ->
+            {ok, Info} = couch_index:get_info(Pid),
+            Sig = IdxState#mrst.sig,
+            DbName = IdxState#mrst.db_name,
+            MrView = lists:keyfind(mrview, 1, IdxState#mrst.views),
```

Comment:
You are making an assumption that the user of the function is only interested only in a very first view out of possibly many.

We could have `view_state(Pid, Name)`. However in this case we need an additional function to list views `list_views(Pid)`. Another option would be to return information about multiple views. 

If we would go the second route the data structure returned from the function would look like (assuming we convert to map)

```
#{
       signature => ...,
       db_name => ...,
       idx_name => ...,
       update_seq => ...,
       purge_seq => ...,
       view_file_path => ...,
       pending_updates => ...,
       views =>
            #{ <view_name> => {
                   id_num => ...,
                   update_seq => ...,
                   purge_seq => ...,
                   map_names => ...,
                   reduce_funs => ...,
                   def => ...,
                   btree_size => btree:size(BTree),
                   options=> ...
             }
          }
       }
},
```

In this case you would need to do extra work in  view_report to print views individually after common info

```
view_report(Pid) ->
    case view_state(Pid) of
        {ok, Report} ->
              {value, {views, Views}, Rest} = lists:keytake(views, 1, Report),
              couch_debug:print_report(Rest),
              lists:map(fun print_view_report/1, Views);
        Error ->
            Error
    end.
```

Maybe we should have both variants:

```
index_state(Pid) -> #{views => #{}}
view_state(IndexState | Pid, ViewName) -> #{}
list_views(IndexState | Pid) -> []
index_report(Pid) when is_pid(Pid) ->
    index_report(index_state(Pid), ViewName);
index_report(State) ->
    ReportWithoutViewsInfo = get it from State,
    couch_debug:print_report(ReportWithoutViewsInfo),
    # views
    Views = get it from State,
    lists:foreach(fun(Name) ->
        view_report(State, Name) end, list_views(State))
    ). 
view_report(Pid, Name) when is_pid(Pid) ->
    view_report(index_state(Pid), Name);
view_report(PidOrState, Name) ->
    couch_debug:print_report(ReportWithoutViewsInfo),
    print_view_report(Pid, Name).

print_view_report(Pid, ViewName) when is_pid(Pid) ->
    print_view_report(index_state(Pid), ViewName);
print_view_report(State, ViewName) ->
    ## View {name}
    couch_debug:print_report(PartivularViewInfo)
    ### map
    {def}
    ### reduce
    {reduce} <--- taken from #mrview.reduce_funs
```

Reply:
I am not sure whether we would need `map_names` key. It could be redundant. As I can see the `<view_name>` in a views map might be sufficient.

Reply:
I have made some progress on viewing reports for multiple views.
```
|                       info                       |                                                                                               value
|                      db_name                     |                                                            shards/00000000-ffffffff/dbv1.1655922592
|                      idx_name                    |                                                                                    _design/dbv1ddoc
|                  pending_updates                 |                                                                                                   0
|                     purge_seq                    |                                                                                                   0
|                     signature                    |                                                                    917d560af42cb62fcaccc52ca8ddcb7d
|                     update_seq                   |                                                                                                   3
|                   view_file_path                 |1/data/.shards/00000000-ffffffff/dbv1.1655922592_design/mrview/917d560af42cb62fcaccc52ca8ddcb7d.view

|                       info                       |                                                                                               value
|                     btree_size                   |                                                                                                  50
|                        def                       |                                                                     function(doc){emit(doc.id, 2);}
|                       id_num                     |                                                                                                   1
|                      options                     |
|                     purge_seq                    |                                                                                                   0
|                    reduce_funs                   |                                                                                        _count
|                     update_seq                   |                                                                                                   3
|                     view_name                    |                                                                                            dbv2view

|                       info                       |                                                                                               value
|                     btree_size                   |                                                                                                  51
|                        def                       |                                                                     function(doc){emit(doc.id, 1);}
|                       id_num                     |                                                                                                   0
|                      options                     |
|                     purge_seq                    |                                                                                                   0
|                    reduce_funs                   |
|                     update_seq                   |                                                                                                   3
|                     view_name                    |                                                                                            dbv1view
```
Going to look into refactoring it into some of the functions you've mentioned.

Reply:
I have added more function clauses for when a `Pid` or an `IdxState` is given. I have also added the ability to retrieve just the index state, the view state, or the index + view state and have done the same regarding printing reports.